wandb/sdk/interface/artifacts.py

```diff
@@ -116,13 +139,11 @@ def download(self, root: Optional[str] = None) -> str:
         """
         raise NotImplementedError
 
-    def ref(self) -> str:
+    def ref_target(self) -> str:
```

Comment:
This is technically a breaking change -- should we do this? To be clear, I much prefer the new name `ref_target`. We'll also need to make sure we update the docs accordingly.

Reply:
This was actually a bit of a problem since the ArtifactManifestEntry has a `ref` attribute and the ArtifactEntry used `ref` as a function call - the underlying usages created mutually exclusive definitions and ultimately typing properties. I chose to modify the one which i thought would minimize any damage. But you are correct, technically could break some user code. thoughts?

Reply:
I think it's unlikely this will impact a meaningful number of users, however it's probably a good idea to give them a better error.  Something like:

```
def ref(self) -> str:
  raise AttributeError("ref has been renamed to ref_target")
```

Reply:
Unfortunately `ref` will never be callable on any of the subclasses. This is because both subclasses assign their own `self.ref` at construction time. So even if we add a method in the interface, it will never be accessible as it is always overrode immediately. Maybe there is a hack around this?